src/wallet/mod.rs

```diff
@@ -3836,6 +3833,52 @@ pub(crate) mod test {
         assert_eq!(details.fee.unwrap_or(0), 250);
     }
 
+    #[test]
+    #[should_panic(expected = "InsufficientFunds")]
+    fn test_bump_fee_unconfirmed_inputs_only() {
+        // We try to bump the fee, but:
+        // - We can't reduce the change, as we have no change
+        // - All our UTXOs are unconfirmed
+        // So, we fail with "InsufficientFunds", as per RBF rule 2:
+        // The replacement transaction may only include an unconfirmed input
+        // if that input was included in one of the original transactions.
+        let (wallet, descriptors, _) = get_funded_wallet(get_test_wpkh());
+        let addr = Address::from_str("2N1Ffz3WaNzbeLFBb51xyFMHYSEUXcbiSoX").unwrap();
+        let mut builder = wallet.build_tx();
+        builder
+            .drain_wallet()
+            .drain_to(addr.script_pubkey())
+            .enable_rbf();
+        let (psbt, mut original_details) = builder.finish().unwrap();
+        // Now we receive one transaction with 0 confirmations. We won't be able to use that for
+        // fee bumping, as it's still unconfirmed!
+        crate::populate_test_db!(
```

Comment:
Would it make sense to test replacing txs with different amounts of inputs (where inputs are of various states of unconfirmed/confirmed)?

Not sure if it should be included for the scope of this PR though...

Reply:
I'm not sure what the "end goal" of such test would be. What are you testing exactly? That we would choose only confirmed inputs for fee bumping?

Reply:
For example, we can test whether BDK allows for RBF of tx with unconfirmed parent.
```
Given:
* Wallet with 2 UTXOs (1 confirmed, 1 unconfirmed).
* We attempt to "drain wallet", resulting in a tx with 2 inputs 
  (1 unconfirmed, 1 confirmed) and 1 output to drain address.

When: 
* Attempt RBF of "drain tx".

Expect:
* Successful RBF of "drain tx".
```

We can also ensure BDK does not pick unconfirmed input (that was not part of original tx):
```
Given:
* Wallet with 2 UTXOs (1 confirmed, 1 unconfirmed).
* We spend the confirmed UTXO to a single output of given amount
  (tx with 1 input, 1 output).

When:
* We attempt RBF.

Expect:
* Fail.
```

Reply:
@danielabrozzoni checking whether we only pick confirmed new inputs would be great too!

Reply:
Right, I'll add it, thanks!

Reply:
@danielabrozzoni Fair, let's skip it :P I like the new test!